src/com/facebook/buck/swift/SwiftDescriptions.java

```diff
@@ -69,4 +60,27 @@ private SwiftDescriptions() {
     output.enableObjcInterop = Optional.of(true);
   }
 
+  static boolean hasSwiftSource(ImmutableSortedSet<SourceWithFlags> srcs) {
+    return Iterables.any(srcs, new Predicate<SourceWithFlags>() {
+      @Override
+      public boolean apply(SourceWithFlags input) {
+        SourcePath sourcePath = input.getSourcePath();
+        return (sourcePath instanceof PathSourcePath) &&
```

Comment:
`srcs` can contain `BuildTargetSourcePath`: it can refer to files that are outputs of other rules, for example when using `genrule` to generate sources.